pandas/_libs/tslibs/conversion.pyx

```diff
@@ -250,6 +250,12 @@ cdef _TSObject convert_to_tsobject(object ts, tzinfo tz, str unit,
         if ts == NPY_NAT:
             obj.value = NPY_NAT
         else:
+            if unit in ["Y", "M"]:
+                # GH#47266 cast_from_unit leads to weird results e.g. with "Y"
+                #  and 150 we'd get 2120-01-01 09:00:00
+                ts = np.datetime64(ts, unit.upper())
```

Comment:
Nit: Is the `upper` needed? The units in the if check are already uppered

Reply:
good point, will update

Reply:
updated+green